### ISSUE TITLE
server-side rendering markup differences

### ISSUE THREAD

**Issue by bmcmahen** [gcanti/tcomb-form#124]
When i use this module to render a form server-side, and then rerun the same code on the client, I get the following warning: 

```
Warning: React attempted to reuse markup in a container but the checksum was invalid. This generally means that you are using server rendering and the markup generated on the server was not what the client was expecting. React injected new markup to compensate which works but you have lost many of the benefits of server rendering. Instead, figure out why the markup being generated is different on the client or server:
 (client) .$name"><label for="2111e75b-23e5-484e-9
 (server) .$name"><label for="3de2ba9c-0384-405b-b
```

It looks like this is because tcomb-form generates random ids for the input fields (for using labels). I suspect this isn't a big deal, but I'm wondering if it's possible to generate consistent (and not entirely random), yet unique ids for these fields, such that it generates the same id on the client and server.

**Comment by gcanti** [comment]
Yeah, as a workaround I'm setting the `id` option for each field. It's a problem for big forms tough. And for lists.
Maybe I could partially fix them:
- for fields: exploiting `this._reactInternalInstance._rootNodeID` (should be consistent)
- for lists: using a counter instead of a uuid

p.s.
Thx for the 400th star :)

**Comment by th0r** [comment]
@gcanti This is not fixed for me. I still have this React warning:

**Comment by gcanti** [comment]
Yeah, I hit this bug as well. Any idea about this? What about an API `Form.rewind()` to be used server side before each rendering? I need a way to tell tcomb-form to reset the counter

**Comment by th0r** [comment]
@gcanti Does every form input has access to it's parent form? You could then initialize this counter in form's constructor and make some kind of `generateFieldId(fieldName)` method, which will prepend this counter with some form-specific string (form component's `key` attribute?).

**Comment by gcanti** [comment]
@th0r I'll do some experiment asap, thanks for the suggestion :+1:

**Comment by snadn** [comment]
maybe use a platform test, such as `typeof window`?
or test the id to Judge it is init or update?

**Comment by gcanti** [comment]
@th0r I'm exploiting `this._reactInternalInstance._rootNodeID` to create a deterministic uid generator. After some tests it seems to work fine. I pushed the code to the master branch, can you please give it a spin with your use case?

**Comment by snadn** [comment]
@gcanti  it work fine in my project now. but the id is different with the reactid, is that ok?
![1](https://cloud.githubusercontent.com/assets/1269858/7645450/f7731f2a-fae7-11e4-85be-77d30a1bf9d0.png)

**Comment by gcanti** [comment]
@snadn it should be ok, actually it's the parent's `reactid`

**Comment by gcanti** [comment]
@snadn it should be ok, actually it's based on the parent's `reactid`.

I've just pushed another version with a shorter generated id.

**Comment by th0r** [comment]
@gcanti Yes, everything works great now!

**Comment by gcanti** [comment]
@th0r @snadn Thanks for your help, much appreciated

Published v0.5.3
